routers/api/v1/repo/issue_label.go

```diff
@@ -0,0 +1,225 @@
+		}
+		return
+	}
+
+	apiLabels := make([]*api.Label, len(updatedIssue.Labels))
+	for i := range updatedIssue.Labels {
+		apiLabels[i] = convert.ToLabel(updatedIssue.Labels[i])
+	}
+
+	ctx.JSON(200, &apiLabels)
+}
+
+func ReplaceIssueLabels(ctx *context.APIContext, form api.IssueLabelsOption) {
+	if !ctx.Repo.IsWriter() {
+		ctx.Status(403)
+		return
+	}
+
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Status(404)
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	var labels []*models.Label
+	if labels, err = loadLabelsByID(form.Labels, issue.RepoID); err != nil {
+		ctx.Error(400, "loadLabelsByID", err)
+		return
+	}
+
+	for i := range issue.Labels {
+		if err := models.DeleteIssueLabel(issue, issue.Labels[i]); err != nil {
+			ctx.Error(500, "NewIssueLabel", err)
+			return
+		}
+	}
+
+	for i := range labels {
+		if !models.HasIssueLabel(issue.ID, labels[i].ID) {
+			if err := models.NewIssueLabel(issue, labels[i]); err != nil {
+				ctx.Error(500, "NewIssueLabel", err)
+				return
+			}
+		}
+	}
+
+	updatedIssue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Status(404)
+		} else {
+			ctx.Error(500, "GetUpdatedIssueByIndex", err)
+		}
+		return
+	}
+
+	apiLabels := make([]*api.Label, len(updatedIssue.Labels))
+	for i := range updatedIssue.Labels {
+		apiLabels[i] = convert.ToLabel(updatedIssue.Labels[i])
+	}
+
+	ctx.JSON(200, &apiLabels)
+}
+
+func DeleteIssueLabel(ctx *context.APIContext) {
+	if !ctx.Repo.IsWriter() {
+		ctx.Status(403)
+		return
+	}
+
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Status(404)
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	label, err := models.GetLabelByID(ctx.ParamsInt64(":id"))
+	if err != nil {
+		if models.IsErrLabelNotExist(err) {
+			ctx.Status(400)
+		} else {
+			ctx.Error(500, "GetLabelByID", err)
+		}
+		return
+	}
+
+	if err := models.DeleteIssueLabel(issue, label); err != nil {
+		ctx.Error(500, "DeleteIssueLabel", err)
+		return
+	}
+
+	ctx.Status(204)
+}
+
+func DeleteAllIssueLabels(ctx *context.APIContext) {
+	if !ctx.Repo.IsWriter() {
+		ctx.Status(403)
+		return
+	}
+
+	issue, err := models.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.Status(404)
+		} else {
+			ctx.Error(500, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	for i := range issue.Labels {
+		if err := models.DeleteIssueLabel(issue, issue.Labels[i]); err != nil {
+			ctx.Error(500, "DeleteIssueLabel", err)
+			return
+		}
+	}
+
+	ctx.Status(204)
+}
+
+func loadLabelsByID(labelIDs []int64, repoID int64) ([]*models.Label, error) {
```

Comment:
What do you think about loadLabelsForIssue or loadCandidateLabelsForIssue, passing the issue instead of repo id? That might be a bit more clear, though the first option could be confused with loading an issue's existing labels and the second is fairly wordy.
Example:
`loadCandidateLabelsForIssue(labelIDs []int64, issue models.Issue) ([]*models.Label, error)`